pkg/security/probe/activity_dump_remote_storage.go

```diff
@@ -45,20 +48,30 @@ func getEndpointURL(endpoint logsconfig.Endpoint, uri string) string {
 
 // ActivityDumpRemoteStorage is a remote storage that forwards dumps to the backend
 type ActivityDumpRemoteStorage struct {
-	urls    []string
-	apiKeys []string
+	urls             []string
+	apiKeys          []string
+	tooLargeEntities map[dump.StorageFormat]map[bool]*atomic.Float64
 
 	client *http.Client
 }
 
 // NewActivityDumpRemoteStorage returns a new instance of ActivityDumpRemoteStorage
 func NewActivityDumpRemoteStorage() (ActivityDumpStorage, error) {
 	storage := &ActivityDumpRemoteStorage{
+		tooLargeEntities: make(map[dump.StorageFormat]map[bool]*atomic.Float64),
 		client: &http.Client{
 			Transport: ddhttputil.CreateHTTPTransport(),
 		},
 	}
 
+	for _, format := range dump.AllStorageFormats() {
+		entities := make(map[bool]*atomic.Float64)
+		for _, compression := range []bool{true, false} {
+			entities[compression] = atomic.NewFloat64(0)
```

Comment:
IMO it would be clearer to use an atomic uintXX here and let the conversion to the sending code

Reply:
👍🏻